### PR TITLE
Rental requesting/checkout

### PR DESCRIPTION
**Pull request by mktraynor**
Rentals controller created

### CODE REVIEW

**Comment by mktraynor** [comment]
Rental Request Form moved from **app/views/items/show** to **app/views/rentals/new.** MVC seems to work fine - but receiving an error with the form much like I had when it was on the show page before Vlad helped last Tuesday. Will have to ask him again this evening :/
Screenshot 2023-02-24 at 5 41 00 PM
Screenshot 2023-02-24 at 5 41 15 PM

**Comment by mktraynor** [comment]
Forgot to attach error message mentioned above
Screenshot 2023-02-24 at 5 40 35 PM

File: app/controllers/rentals_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  def find_item
+    @item = Item.find(params[:item_id])
+  end
+
+  def find_rental
+    @rental = Rental.find(params[:id])
+  end
+
+  def rental_params
+    params.require(:rental).permit(:start_date, :end_date, :confirmed, :user_id, :item_id)
```
Comment (elizaup) [inline app/controllers/rentals_controller.rb]:
```suggestion
    params.require(:rental).permit(:start_date, :end_date, :user_id, :item_id)
```

Comment (elizaup) [reply app/controllers/rentals_controller.rb]: we don't have confirmed in our model so this should not be here

File: app/controllers/rentals_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  before_action :find_rental, only: [ :confirmation, :request, :show ]
+  before_action :find_item, only: [ :confirmation, :request ]
+
+  def show
+
+  end
+
+  def create
+    @rental = Rental.new(rental_params)
+    authorize @rental
```
Comment (elizaup) [inline app/controllers/rentals_controller.rb]:
```suggestion
```

Comment (elizaup) [reply app/controllers/rentals_controller.rb]: don't have pundit set up for this page, so no need to add this here now

File: app/controllers/rentals_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    else
+      render 'request', status: :unprocessable_entity
+    end
+  end
+
+
+  def rental_request
+  end
+
+  def confirmation
```
Comment (elizaup) [inline app/controllers/rentals_controller.rb:24]: do we have this concept though confirmed?

File: app/controllers/rentals_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+class RentalsController < ApplicationController
+  before_action :find_rental, only: [ :confirmation, :request, :show ]
+  before_action :find_item, only: [ :confirmation, :request ]
+
```
Comment (elizaup) [inline app/controllers/rentals_controller.rb:4]: you are missing a def new method that corresponds with rental/new.html.erb

File: config/routes.rb [hunk 5]
```diff
@@ -27,7 +27,8 @@
   # Profile
   # get "/profile", to: "profile#show"
 
-  # Rental "checkout"
-  # get "/items/rental/new", to: "rental#new"
-  # post "items/rental", to: "rental#create"
+  # New / Rental (Request)
+  get "rentals/new", to: "rentals#new"
```
Comment (elizaup) [inline config/routes.rb:31]:
```suggestion

```

Comment (elizaup) [inline config/routes.rb:31]:
```suggestion

```

File: config/routes.rb [hunk 6]
```diff
@@ -27,7 +27,8 @@
...
+  post "rentals/new", to: "rentals#create" #?? ask TA
```
Comment (elizaup) [inline config/routes.rb:32]:
```suggestion
  post "rentals/new", to: "rentals#create" #?? ask TA
```

Comment (elizaup) [reply config/routes.rb:32]: It seems your routes are not updated with what Vlad told us to do:


```
Rails.application.routes.draw do
  devise_for :users
  # Homepage where you can see all clothes
  root "items#index"
  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html

  # Defines the root path route ("/")
  # root "articles#index"\
  resources :items, only: [:new, :create, :show, :update, :destroy] do
    resources :rentals, only: [:create]
  end

  # New / Items
  # get "items/new", to: "items#new", as: :new_item
  # post "items", to: "items#create"

  # Show / Items
  # get "items/:id", to: "items#show", as: :item

  # Edit / Items
  get "items/:id/edit", to: "items#show", as: :edit_item
  # patch "items/:id", to: "items#update"

  # Destroy / Items
  # delete "items/:id", to: "items#destroy"

  # Profile
  # get "/profile", to: "profile#show"

  # Rental "checkout"
  # get "/items/rental/new", to: "rental#new"
  # post "items/rental", to: "rental#create"
end

```

File: app/controllers/rentals_controller.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      render 'request', status: :unprocessable_entity
+    end
+  end
+
+
+  def rental_request
+  end
+
+  def confirmation
+    if @rental.update(confirmed)
```
Comment (elizaup) [inline app/controllers/rentals_controller.rb:25]: we don't have update in our rental model so this does not make sense

**Review by elizaup — CHANGES_REQUESTED** [review]

**Pull request closed without merging**